npm/pkg/dataplane/policies/chain-management_linux.go

```diff
@@ -358,12 +403,25 @@ func (pMgr *PolicyManager) allCurrentAzureChains() ([]string, error) {
 		return nil, nil
 	}
 	lines := strings.Split(string(searchResults), "\n")
+	if len(lines) == 1 && lines[0] == "" {
+		// this should never happen: gotMatches is true, but there is no content in the searchResults
+		return nil, errInvalidGrepResult
+	}
+	lastIndex := len(lines) - 1
+	lastLine := lines[lastIndex]
+	if lastLine == "" {
+		// remove the last empty line (since each line ends with a newline)
+		lines = lines[:lastIndex]
```

Comment:
instead of resizing the while array, could we just check if line != "" in the for loop below ?

Reply:
looked into this while implementing, and golang slices reference arrays under the hood. This operation will just update the slice length (just an integer <= array length), and not impact memory allocations for the array.

Let me add a short comment about this